crawler/plugins/emitters/base_http_emitter.py

```diff
@@ -0,0 +1,74 @@
+import logging
+import time
+
+import requests
+
+from iemit_plugin import IEmitter
+
+logger = logging.getLogger('crawlutils')
+
+
+class BaseHttpEmitter:
+
+    def get_emitter_protocol(self):
+        raise NotImplementedError("Use http or https emitter plugin instead.")
+
+    def init(self, url, timeout=1, max_retries=5, emit_format='csv'):
+        IEmitter.init(self, url,
+                      timeout=timeout,
+                      max_retries=max_retries,
+                      emit_format=emit_format)
+        if emit_format == 'json':
+            self.emit_per_line = True
+
+    def emit(self, frame, compress=False,
+             metadata={}, snapshot_num=0):
+        """
+
+        :param frame: a frame containing extracted features
+        :param compress:
+        :param metadata:
+        :param snapshot_num:
+        :return: None
+        """
+        iostream = self.format(frame)
+        if compress:
+            proto = self.get_emitter_protocol()
+            raise NotImplementedError(
+                '%s emitter does not support gzip.' % proto
+            )
+        if self.emit_per_line:
+            iostream.seek(0)
+            for line in iostream.readlines():
+                self.post(line, metadata)
+        else:
+            self.post(iostream.getvalue(), metadata)
+
+    def post(self, content='', metadata={}):
+        headers = {'content-type': 'application/csv'}
+        for attempt in range(self.max_retries):
+            try:
+                response = requests.post(self.url, headers=headers,
+                                         params=metadata,
```

Comment:
if you just remove the `params` parameter you'll address pr #259's issue #251 as well :)